lib/instrumentation/modules/http2.js

```diff
@@ -141,7 +147,7 @@ module.exports = function (http2, agent, { enabled }) {
 
   function updateHeaders (headers) {
     var trans = agent._instrumentation.currTransaction()
-    if (trans) {
+    if (trans && !trans.ended) {
```

Comment:
FYI -- I didn't notice any difference in the behavior of the test suite with the additional `trans.ended` check here.  Presuming its here to skip the extra work if the work isn't needed.

Reply:
This is a case similar to this guard lower down on `wrapEnd`:
https://github.com/elastic/apm-agent-nodejs/blob/f6c4ee8a978b6edfee31b74f2e410c1a86206b38/lib/instrumentation/modules/http2.js#L171-L177

There is actually a possible crash here:

```
/Users/trentm/el/apm-agent-nodejs8/lib/instrumentation/modules/http2.js:149
      trans.res.statusCode = status
                           ^

TypeError: Cannot set properties of null (setting 'statusCode')
    at updateHeaders (/Users/trentm/el/apm-agent-nodejs8/lib/instrumentation/modules/http2.js:149:28)
    at ServerHttp2Stream.<anonymous> (/Users/trentm/el/apm-agent-nodejs8/lib/instrumentation/modules/http2.js:159:7)
    at Timeout._onTimeout (/Users/trentm/el/apm-agent-nodejs8/play-http2-insecure-server.js:27:12)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
```

...if the server calls stream.respond() *after* the stream has been destroyed (which ends the transaction, which nulls out `trans.res`).

However, I (lazily) didn't add a test case for this. It was an unrelated crasher I hit when initially attempting to repro the issue (#2670).